### PR TITLE
feat: added loading animations where necessarry

### PR DESCRIPTION
**Pull request by rbadici-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/web/pages/category/[slug].vue [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-    >
-      <template #sidebar>
-        <CategoryTree :category="category" />
-        <CategorySorting />
-        <CategoryItemsPerPage class="mt-6" :total-products="productsCatalog.pagination.totals" />
-        <CategoryFilters :facets="productsCatalog.facets" />
-      </template>
-    </CategoryPageContent>
-  </NuxtLayout>
+  <div class="relative" :class="{ 'pointer-events-none opacity-50': loading }">
```
Comment (maxiroellplenty) [inline apps/web/pages/category/[slug].vue]: loading spinner should only be on the product page not the whole page

**Review by maxiroellplenty — DISMISSED** [review]
The loading spinner should only be on the part of the page that is actually loading.
Also its missing, loading spinner for search.

**Comment by Tim-M-S** [comment]
Changing the filters or sorting on the category page should only reload the item grid
image

**Comment by Tim-M-S** [comment]
Changing the shipping or payment provider should show a loader on both of them
image

**Comment by Tim-M-S** [comment]
Switching between the tabs in the myaccount should show a loader in the changing section
image

**Review by FabianGerke — APPROVED** [review]

**Pull request merged**